src/Core/Framework/DataAbstractionLayer/Validation/EntityExists.php

```diff
@@ -35,10 +35,7 @@ class EntityExists extends Constraint
      */
     public function __construct(array $options)
     {
-        $options = array_merge(
-            ['criteria' => new Criteria()],
-            $options
-        );
+        $options['criteria'] ??= new Criteria();
```

Comment:
Could you explain why this change is necessary? I agree your change is for better performance, but Is it not the root cause of the bug?